lang/csharp/src/apache/main/CodeGen/CodeGen.cs

```diff
@@ -57,14 +57,6 @@ public class CodeGen
         /// </value>
         public IList<Protocol> Protocols { get; private set; }
 
-        /// <summary>
-        /// Gets mapping of Avro namespaces to C# namespaces.
-        /// </summary>
-        /// <value>
-        /// The namespace mapping.
-        /// </value>
-        public IDictionary<string, string> NamespaceMapping { get; private set; }
```

Comment:
Isn't this potentially API breaking to remove this?

Reply:
It is a good call out. I reshuffled the code and added back the NamespaceMapping property and marked it obsolete.
Thebiggest issue with this serious bug is that the best solution is to change the schema text before the actual parsing, otherwise the whole already created Schema or Protocol object must be recreated. The documentation might need mapping. So there might be some actual breaking for some people who generate code on the fly and not via avrogen, however using a dynamically compiled code is not very likely since using the generated SpecificRecord is not very convinient.